sources/packages/backend/apps/api/src/route-controllers/supporting-user/supporting-user.supporting-users.controller.ts

```diff
@@ -216,7 +226,9 @@ export class SupportingUserSupportingUsersController extends BaseController {
 
     // Ensure that the user providing the supporting data is not the same user that
     // submitted the Student Application.
-    if (application.student.user.userName === userToken.userName) {
+    if (
+      supportingUser.application.student.user.userName === userToken.userName
+    ) {
```

Comment:
Please also validate `supportingUser.supportingData` to check if the supporting data has already been provided for the supporting user.

![image](https://github.com/user-attachments/assets/280df1fc-5520-4c52-8716-eef29d8858ba)